### DISCUSSION TITLE
Orin NX set recovery USB port to host mode instead of OTG mode

### DISCUSSION THREAD

**Discussion by pboucherat-thegoodpenguin** [OE4T/meta-tegra#1645] [Q&A] [answered]
I want the recovery USB USB2-0 to be set to host mode for normal operation, but I think it needs to be in OTG mode for initrd-flash to work - am I actually right about that or is it ok to have it in host mode? If it's not ok, then I was thinking to modify the initrd-flash script to use a slightly different dtb file for RCM boot, i.e. this dtb would set OTG mode on USB2-0 but otherwise be the same as the actual kernel dtb. Is there any reason why this wouldn't work? Or is there a way to do this already?

**Comment by dwalkes** [comment]
There's a relevant PR at https://github.com/OE4T/meta-tegra/pull/1394 for r32.7.x but I don't know if something like that exists on Jetpack 6 branches yet.

**Reply by pboucherat-thegoodpenguin** [reply]
Ah, thanks, that looks useful! Need to improve my googling skills :) I'm on a Jetpack 5 branch so that's fine.

**Reply by dwalkes** [reply]
I don't think it's on Jetpack 5 branches either, probably needs to be cherry-picked and tested on the appropriate branch.

**Reply by pboucherat-thegoodpenguin** [reply]
Yes, I can see that, but it still looks like it's doable, will just patch manually.

**Reply by pboucherat-thegoodpenguin** [reply, marked as answer]
I made a start on this, but didn't finish as I discovered that USB 3 devices do work on the recovery port on my board, but the cable I was using was making my USB 3.1 camera drop down to USB 2.1, using a different USB-C cable fixed the problem.

**Reply by pboucherat-thegoodpenguin** [reply]
Not really the answer I know.
